developer-hub/src/reference_docs/glossary.rs

```diff
@@ -36,27 +36,84 @@
+//! without bloating the blockchain state.
+//!
+//! #### Host Functions:
+//!
+//! Host functions are the node's API, these are functions provided by the runtime environment (the
+//! [host](#host)) to the Wasm runtime. These functions allow the Wasm code to interact with and
+//! perform operations on the [node](#node), like accessing the blockchain state.
 //!
 //! #### Runtime API:
 //!
+//! This is the API of the runtime, it acts as a communication bridge between the runtime and the
+//! node, serving as the exposed interface that facilitates their interactions.
+//!
 //! #### Dispatchable:
 //!
-//! Callable
+//! Dispatchables are [function objects](https://en.wikipedia.org/wiki/Function_object) that act as
+//! the entry points in [FRAME](frame) pallets. They can be called by internal or external entities
+//! to interact with the blockchain's state. They are a core aspect of the runtime logic, handling
+//! transactions and other state-changing operations.
+//!
+//! **Synonyms**: Callable
 //!
 //! #### Extrinsic
 //!
+//! An extrinsic is a general term for a piece of data that is originated outside of the runtime,
+//! included into a block and leads to some action. This includes user-initiated transactions as
+//! well as inherents which are placed into the block by the block-builder.
 //!
 //! #### Pallet
 //!
+//! Similar to software modules in traditional programming, [FRAME](frame) pallets in Substrate are
+//! modular components that encapsulate distinct functionalities or business logic. Just as
+//! libraries or modules are used to build and extend the capabilities of a software application,
+//! pallets are the foundational building blocks for constructing a blockchain's runtime with frame.
+//! They enable the creation of customizable and upgradeable networks, offering a composable
+//! framework for a Substrate-based blockchain. Each pallet can be thought of as a plug-and-play
+//! module, enhancing the blockchain's functionality in a cohesive and integrated manner.
+//!
 //! #### Full Node
 //!
+//! It is a node that prunes historical states, keeping only recent finalized block states to reduce
+//! storage needs. Full nodes provide current chain state access and allow direct submission and
+//! validation of extrinsics, maintaining network decentralization.
+//!
 //! #### Archive Node
 //!
+//! An archive node is a specialized node that maintains a complete history of all block states and
+//! transactions. Unlike a full node, it does not prune historical data, ensuring full access to the
+//! entire blockchain history. This makes it essential for detailed blockchain analysis and
+//! historical queries, but requires significantly more storage capacity.
+//!
 //! #### Validator
 //!
+//! A validator is a node that participates in the consensus mechanism, validating transactions and
+//! blocks, and maintaining the integrity and security of the network.
```

Comment:
I think block production is worth an extra mention here.
```suggestion
//! A validator is a node that participates in the consensus mechanism of the network. 
//! Its role includes block production,  transaction validation, and network integrity and security maintenance.
```